core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -2157,6 +2165,23 @@ private static RelDataType deriveTypeMapFromEntries(SqlOperatorBinding opBinding
           OperandTypes.NUMERIC,
           SqlFunctionCategory.NUMERIC);
 
+  /** The "LOG10(numeric)" function. Returns the base 10 logarithm of numeric. */
+  @LibraryOperator(libraries = {MYSQL, SPARK})
+  public static final SqlFunction LOG10_MS =
+      SqlBasicFunction.create("LOG10",
+          ReturnTypes.DOUBLE_FORCE_NULLABLE,
+          OperandTypes.NUMERIC,
+          SqlFunctionCategory.NUMERIC);
```

Comment:
I have a question, `LOG10` and `LN` have been defined in `SqlStdOperatorTable`. Defining functions with the same name and Sqlkind here may cause `SqlOperatorTable#lookupOperatorOverloads` to find multiple `overloads`, will this cause SQL validation exceptions?

Reply:
Thank you for your feedback. I have asked similar questions in jira. Because the behavior of log10 and ln functions is inconsistent between mysql and spark and the log10 and ln functions in SqlStdOperatorTable. The problem you mentioned does occur. I am trying to find a reasonable way to solve this problem. If you have a way to solve this problem without modifying SqlStdOperatorTable, that would be great.

Reply:
The problem I am currently encountering is that ReturnTypes.DOUBLE_FORCE_NULLABLE of LOG10_MS does not take effect.

Reply:
You can try to change the function `SqlKind`  in `SqlLibraryOperators`, because the default `equals` method of `SqlOperator` is to compare function name and SqlKind.